optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -147,10 +151,99 @@
      * }
      */
 
-    // TODO
-    // join(..., planningVariableContainsCached(Talk::getPeriod, (Period a, Period b) -> a.overlaps(b)))
-    // get the period value range, does a cartesian product on it, so it maps every period to an overlapping periodList
-    // then keep an index from every period to all talks in an overlapping period (possible the same period)
+    /**
+     * For pairs of intervals [a,b) and [c,d), returns only those pairs where [a,b) and [c,d) overlap (or, in other
+     * words, there exists an x such that x in [a,b) and x in [c,d)).
+     * These are exactly the pairs where {@code a < d} and {@code b > c} (or in other words, when the first interval
+     * starts before the second one ends and the first interval ends after the second interval begins).
+     * 
+     * @param <A> the type of the first argument
+     * @param <B> the type of the second argument
+     * @param <Property_> the type used to define the interval, comparable
+     * 
+     * @param leftStartMapping maps the first argument to the first interval start point
+     * @param leftEndMapping maps the first argument to the first interval end point
```

Comment:
```suggestion
     * @param leftStartMapping maps the first argument to the first interval start point (inclusive)
     * @param leftEndMapping maps the first argument to the first interval end point (exclusive)
```

The method description marks the intervals `[a, b)`, which is the correct mathematical notation for an interval open on the right, but let's make it crystal clear.